client/container_inspect.go

```diff
@@ -19,10 +19,10 @@ func (cli *Client) ContainerInspect(ctx context.Context, containerID string) (ty
 	if err != nil {
 		return types.ContainerJSON{}, wrapResponseError(err, serverResp, "container", containerID)
 	}
+	defer ensureReaderClosed(serverResp)
```

Comment:
hm, I'm a bit on the fence to change this if there's no specific reason (from your PR description, I _think_ you're trying to make the code consistent with `ContainerInspectWithRaw`?).

Although the overhead of `defer` has become a lot less in recent versions of Go; if there's no real need (or it would substantially improve readability), "not" using `defer` is probably still the correct approach.

Reply:
Sure, that makes sense. I thought possibly there was some inconsistency regarding writing these up. I had a look through the edit history of the file and found nothing that suggested there was an intentional decision to have it defined as such.

Correct, I am making the code consistent with `ContainerInspectWithRaw`.

Reply:
Not to nitpick, though defer actually incur s small cost compared with the non-defer version.

The ContainerInspectWithRaw uses defer, likely because there are two returns.

I personally would be fine either way, though.

Reply:
@yongtang As this is merely cosmetic, it is a matter of taste. I will leave it as a choice to the Maintainers of the project.